app/javascript/packs/document-capture.jsx

```diff
@@ -42,11 +48,15 @@ loadPolyfills(['fetch']).then(() => {
         }}
       >
         <I18nContext.Provider value={i18n.strings}>
-          <AssetContext.Provider value={assets}>
-            <DeviceContext.Provider value={device}>
-              <DocumentCapture isLivenessEnabled={isLivenessEnabled} />
-            </DeviceContext.Provider>
-          </AssetContext.Provider>
+          <ServiceProviderContext.Provider
+            value={serviceProvider.name ? serviceProvider : undefined}
```

Comment:
this line made me question the structure a bit, like at first I thought it was a backwards null check (ex it could have been`serviceProvider?.name`

But what if we just passed in two separate SP properties to the context? `serviceProviderName` `serviceProviderBackToSPUrl`? Or do we expect the SP to grow in complexity over time?

Reply:
> this line made me question the structure a bit, like at first I thought it was a backwards null check (ex it could have been`serviceProvider?.name`
> 
> But what if we just passed in two separate SP properties to the context? `serviceProviderName` `serviceProviderBackToSPUrl`? Or do we expect the SP to grow in complexity over time?

I don't have a strong feeling one way or the other. The whole thing is a bit confusing in how `name` alone is used as a condition for whether we reference any properties of the service provider. I'd actually tinkered with pulling out a function here, like:

```jsx
function getServiceProvider() {
  const name = appRoot.getAttribute('data-sp-name');
  const failureToProofURL = appRoot.getAttribute('data-failure-to-proof-url');
  if (name && failureToProofURL) {
    return { name, failureToProofURL };
  }
}

// ...

<ServiceProviderContext.Provider value={getServiceProvider()} />
```

Passing these as individual props avoids needing to make this condition here, but ultimately it just pushes the responsibility for checking that to the consumer of the context. With the current implementation, this exists by checking `serviceProvider` as possibly undefined, and would become checking `serviceProvider.name`. I guess it could be nice since the condition is necessary anyways, and possibly some usefulness in picking some or the other of these properties even if one is set and the other is not? But I think if we consider these as a collective unit of properties, then from a consumer's point of view we should want for the full set to either exist or not exist in its entirety. This is unlike `AcuantProvider` or `UploadContextProvider`, where some of props could potentially be undefined, and the context can still make sense.

tl;dr: I lean to thinking that passing a single value would be best, though it could be cleaned up to address the readability concern you mentioned, possibly using an implementation like in the snippet above.

Reply:
Yeah that function that checks both properties definitely makes it clearer to me, let's do that!

Reply:
> Yeah that function that checks both properties definitely makes it clearer to me, let's do that!

Updated in 8e883b447.